Sprint 05

KamilaSalimzhanova opened the pull request:
Думала, надо ли сохранять totalAccuracy в userDefaults, если нет, то как его можно посчитать с функции store в statisticserviceimplementation. Если просто оставить как var totalAccuracy: Double -> ошибка: class has no initialisers

KamilaSalimzhanova committed:
Добавила слабые ссылки в замыкания System alert и dispatch

KamilaSalimzhanova committed:
Added lines for clarification

KamilaSalimzhanova committed:
Added factory protocol -> abstraction, composition, debugging

KamilaSalimzhanova committed:
Delegate for QuestionFactory

KamilaSalimzhanova committed:
Added naming of sections in viewController

KamilaSalimzhanova committed:
modified questionFactory

KamilaSalimzhanova committed:
Added delegate for AlertPresenter

KamilaSalimzhanova committed:
completed statisticsService with userDefaults and parsed incoming data

So1ovyov commented:
@KamilaSalimzhanova привет! Не совсем понял вопрос, totalAccuracy ты как я вижу сохраняешь и в store используешь корректно.

So1ovyov commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -118,42 +84,54 @@ final class MovieQuizViewController: UIViewController {
+        let bestGame = statisticService.bestGame
+        let gamesCountPlayed = "Количество сыгранных игр: \(statisticService.gamesCount)"
+        let bestGameRecord = "Рекорд: \(bestGame.correct)\\\(bestGame.total) (\(bestGame.date.dateTimeString))"
+        let averageAccuracy = "Средняя точность: \(String(format: "%.2f", statisticService.totalAccuracy))%"
```
Чтобы формат был без лишних нулей после запятой, предлагаю сделать так:
```
let averageAccuracy = "Средняя точность: \(String(format: "%.0f%", statisticService.totalAccuracy))%"
```

So1ovyov commented on MovieQuiz/Presentation/StatisticService.swift:
```diff
@@ -0,0 +1,73 @@
+    }
+    
+    func store(correct count: Int, total amount: Int) {
+        self.totalAccuracy = Double(count)/Double(amount) * 100
```
Здесь расчет верный, никаких ошибок не воспроизводится

So1ovyov commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -118,42 +84,54 @@ final class MovieQuizViewController: UIViewController {
+    // MARK: - Actions
+    
     @IBAction private func noButtonClicked(_ sender: Any) {
         noButton.isEnabled = false
```
Чтобы во всем коде не дублировать эти строки, предлагаю создать приватный метод для изменения состояния кнопок:
```
    private func changeButtonState(isEnabled: Bool) {
        noButton.isEnabled = isEnabled
        yesButton.isEnabled = isEnabled
    }
```

KamilaSalimzhanova committed:
Updated based on review:
1.  private function to decrease duplicates
2. format \(String(format: "%.0f%", statisticService.totalAccuracy))
